nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/ScanHBase.java

```diff
@@ -0,0 +1,562 @@
+                ){
+                flowFile = session.putAttribute(flowFile, "scanhbase.results.found", Boolean.toString(handler.isHandledAny()));
+                session.transfer(flowFile, REL_ORIGINAL);
+                session.commit();
+                return;
+            }
+
+            if (hangingRows != null && !hangingRows.isEmpty()) {
+                FlowFile lastFF = session.create(flowFile);
+                final Map<String, String> attributes = new HashMap<>();
+                attributes.put(HBASE_TABLE_ATTR, tableName);
+                attributes.put(HBASE_ROWS_COUNT_ATTR, Long.toString(rowsPulledHolder.get()));
+                attributes.put(CoreAttributes.MIME_TYPE.key(), "application/json");
+                attributes.put(HBASE_ROWS_COUNT_ATTR, Long.toString(hangingRows.size()));
+                lastFF = session.putAllAttributes(lastFF, attributes);
+
+                final AtomicReference<IOException> ioe = new AtomicReference<>(null);
+                lastFF = session.write(lastFF, (out) -> {
+                    for (Iterator<Tuple<byte[], ResultCell[]>> iter = hangingRows.iterator(); iter.hasNext();){
+                        Tuple<byte[], ResultCell[]> r = iter.next();
+                        serializer.serialize(r.getKey(), r.getValue(), out);
+                        if (iter.hasNext()){
+                            out.write(nl);
+                        }
+                    }
+                });
+
+                Relationship rel = REL_SUCCESS;
+                IOException error = ioe.get();
+                if (error != null){
+                    lastFF = session.putAttribute(lastFF, "scanhbase.error", error.toString());
+                    rel = REL_FAILURE;
+                }
+                session.transfer(lastFF, rel);
+                flowFile = session.putAttribute(flowFile, "scanhbase.results.found", Boolean.toString(handler.isHandledAny()));
+                session.transfer(flowFile, REL_ORIGINAL);
+            }
+            session.commit();
+
+        }catch (final Exception e) {
+            getLogger().error("Failed to receive data from HBase due to {}", e);
+            session.rollback();
+            // if we failed, we want to yield so that we don't hammer hbase.
+            context.yield();
+        }
+    }
+
+    /**
+     * @param columnsValue a String in the form colFam:colQual,colFam:colQual
+     * @return a list of Columns based on parsing the given String
+     */
+    private List<Column> getColumns(final String columnsValue) {
+        final String[] columns = (columnsValue == null || columnsValue.isEmpty() ? new String[0] : columnsValue.split(","));
+
+        List<Column> columnsList = new ArrayList<>(columns.length);
+
+        for (final String column : columns) {
+            if (column.contains(":"))  {
+                final String[] parts = column.split(":");
+                final byte[] cf = parts[0].getBytes(StandardCharsets.UTF_8);
```

Comment:
Shouldn't this charset be configurable? Same for the others here.

Reply:
A charset for data presentation is configurable. It's defined in onScheduled while creating serializer.
A charset for hbase rowkey and column family/qualifier names is utf-8 intentionally to be consistent with GetHBase and FetchHBase.
I would prefer to open new JIRA to handle charset used for row key and column family/qualifiers throughout the nifi-hbase package.